### DISCUSSION TITLE
What does send_time represents?

### DISCUSSION THREAD

**Discussion by keshav-Gupta-IS** [psycopg/psycopg2#1555] [Q&A]
I am referring to this documentation https://www.psycopg.org/docs/extras.html

I am confused about this block of documentation specifically

class psycopg2.extras.ReplicationMessage
A replication protocol message.

payload
The actual data received from the server.

An instance of either bytes() or unicode(), depending on the value of decode option passed to [start_replication()](https://www.psycopg.org/docs/extras.html#psycopg2.extras.ReplicationCursor.start_replication) on the connection. See [read_message()](https://www.psycopg.org/docs/extras.html#psycopg2.extras.ReplicationCursor.read_message) for details.

**send_time**
A [datetime](https://docs.python.org/3/library/datetime.html#module-datetime) object representing the server timestamp at the moment when the message was sent.

What do we mean by the server here is it application server or DB server ?

What does this timestamp represents is it the time the record was present in wal_logs or the time we were able to read it?

Any help would be appreciated!!

**Comment by keshav-Gupta-IS** [comment]
Hi can anyone please help here. I am asking this question because I am seeing the difference of about 5 -10 minutes between send_time and the record commit time.

**Reply by dvarrazzo** [reply]
I think you should see the matching features in the postgres documentation. I am not overly familiar with the replication layer, so can't advise.

**Reply by keshav-Gupta-IS** [reply]
Thanks!! I will go over the postgres documentation.
